presto-hive/src/main/java/com/facebook/presto/hive/HiveClient.java

```diff
@@ -1047,7 +1051,13 @@ else if (value instanceof Slice) {
                 }
                 else {
                     checkArgument(value instanceof Boolean || value instanceof Double || value instanceof Long, "Only Boolean, Slice (UTF8 String), Double and Long partition keys are supported");
-                    filter.add(value.toString());
+                    if (assumeCanonicalPartitionKeys) {
+                        filter.add(value.toString());
+                    }
+                    else {
+                        // Hive treats '0', 'false', and 'False' the same. However, the metastore differentiates between these
+                        filter.add(PARTITION_VALUE_WILDCARD);
```

Comment:
All the cursors use `booleanPartitionKey` which only works for `true` / `false` (case insensitive). `bigintPartitionKey` should work though because it just calls `Long.parseLong`. We should see how Hive treats values other than `0` and `1`.